crates/scarb-api/src/lib.rs

```diff
@@ -204,15 +204,20 @@ pub fn package_matches_version_requirement(
     name: &str,
     version_req: &VersionReq,
 ) -> Result<bool> {
-    let packages: Vec<&scarb_metadata::PackageMetadata> = metadata
+    let mut packages = metadata
         .packages
         .iter()
-        .filter(|package| package.name == name)
-        .collect();
-    match packages[..] {
-        [] => Ok(true),
-        [package] => Ok(version_req.matches(&package.version)),
-        _ => Err(anyhow!("Package {name} is duplicated in dependencies")),
+        .filter(|package| package.name == name);
+
+    let package = packages.next();
+
+    if packages.next().is_some() {
+        return Err(anyhow!("Package {name} is duplicated in dependencies"));
+    }
```

Comment:
This can get error prone pretty fast, I don't think collecting to vector hurts here